File: 092/92.py
#!/usr/bin/python

def f(n):
    return n**2 if n < 10 else (n % 10)**2 + f(int(n / 10))

count = 0
for n in range(1, 10000000):
    print (n)
    a = n
    while a != 1 and a != 89:
        a = f(a)
    if a == 89:
        count += 1

print (count)

File: 092/92.c
#include <stdio.h>

int f(int n)
{
    return n < 10 ? (n * n) : ((n % 10) * (n % 10) + f(n / 10));
}

int main()
{
    int count = 0;
    int a, b;
    for (a = 1; a < 10000000; a++) {
        b = a;
        while (b != 1 && b != 89) b = f(b);
        if (b == 89) count++;
    }

    printf("%d\n", count);
}
